frontend/src/Console.jsx

```diff
@@ -36,26 +47,70 @@ function Console(props) {
   }
 
   function toggleModal() {
-    
     setModal({ isOpen: !modalState.isOpen });
   }
 
+  function sendForm(formDetails) {
+    if (formInfo.postOrPut === "post") {
+      apiUtil[formInfo.postOrPut](
+        `/${formInfo.entityName}`,
+        formDetails,
+        (data) => {
+          setDishes([data]);
+          console.log(data);
+        },
+        (err) => {
+          console.log(err);
+        }
+      );
+    } else {
+      apiUtil[formInfo.postOrPut](
+        `/${formInfo.entityName}/${formInfo.entity._id}`,
+        formDetails,
+        (data) => {
+          setDishes([data]);
+          console.log(data);
+        },
+        (err) => {
+          console.log(err);
+        }
+      );
+    }
+  }
+
+  function toggleLanguage() {
+    if (language === "hebrew") {
+      localStorage.setItem("language", "english");
+    } else {
+      localStorage.setItem("language", "hebrew");
+    }
+    setLanguage(localStorage.language);
+  }
+
   return (
     <>
-        <userContext.Provider
-          value={{
-            user,
-            auth,
-            admin,
-            currentUrl,
-            setUrl,
-            setAuthContext: setUserAndAuth,
-            isModalOpen: modalState.isOpen,
-            toggleModal: toggleModal
-          }}
-        >
-          <Page  />
-        </userContext.Provider>
+      <userContext.Provider
+        value={{
+          user,
+          auth,
+          isAdmin,
+          currentUrl,
+          setUrl,
+          setAuthContext: setUserAndAuth,
+          isModalOpen: modalState.isOpen,
+          toggleModal,
+          language,
+          toggleLanguage,
+        }}
+      >
+        <Page
+          formInfo={formInfo}
```

Comment:
Looks good!
The first 3 params are generic, yet the last two are specific.
So maybe put the last 2 into a special variable (like `additionalProps` which will be an object that you construct based on the current url). Also, even though it's accessible from the context, I think it's more clear if you pass the `currentUrl` here as props to `Page` and then inside `Page` render whatever needed based on that

Reply:
We can also just pass everything via the context - how did you choose what to pass in the context and what to pass directly to `Page`? we can think about it again

Reply:
I really didn't have any logic behind it. We should think whats better for what

Reply:
so I'll have a function in a different file, similar to fields, that will return whatever additional props are needed for every URL?

Reply:
Yeah, sounds good